File: desktop/src/com/inkostilation/pong/desktop/display/shapes/AbstractShape.java
package com.inkostilation.pong.desktop.display.shapes;

import com.badlogic.gdx.Gdx;

import java.util.ArrayList;
import java.util.List;

public abstract class AbstractShape implements IShape {

    private List<IShape> children = new ArrayList<>();

    private boolean ready = false;

    @Override
    public void addChild(IShape shape) {
        children.add(shape);
    }

    @Override
    public List<IShape> getChildren() {
        return children;
    }

    @Override
    public DrawRect getChildrenRect() {
        return new DrawRect(new Position(0, 0), new Position(Gdx.graphics.getWidth(), Gdx.graphics.getHeight()));
    }

    @Override
    public boolean isReady() {
        return ready;
    }

    public void setReady(boolean ready) {
        this.ready = ready;
    }
}

File: desktop/src/com/inkostilation/pong/desktop/display/shapes/AbstractShape.kt
package com.inkostilation.pong.desktop.display.shapes

import com.badlogic.gdx.Gdx
import java.util.*

abstract class AbstractShape : IShape {
    private override val children: MutableList<IShape> = ArrayList()
    override var isReady: Boolean = false public get() {
        return field
    }

    override fun addChild(shape: IShape) {
        children.add(shape)
    }

    override fun getChildren(): List<IShape> {
        return children
    }

    override val childrenRect: IShape.DrawRect
        get() = IShape.DrawRect(IShape.Position(0, 0), IShape.Position(Gdx.graphics.width.toFloat(), Gdx.graphics.height.toFloat()))

}